res/controllers/Hercules-DJControl-Inpulse-300-script.js

```diff
@@ -0,0 +1,234 @@
+};
+
+DJCi300._convertWheelRotation = function(value) {
+    // When you rotate the jogwheel, the controller always sends either 0x1
+    // (clockwise) or 0x7F (counter clockwise). 0x1 should map to 1, 0x7F
+    // should map to -1 (IOW it's 7-bit signed).
+    return value < 0x40 ? 1 : -1;
+};
+
+// The touch action on the jog wheel's top surface
+DJCi300.wheelTouch = function(channel, control, value, _status, _group) {
+    var deck = channel;
+    if (value > 0) {
+        //  Touching the wheel.
+        if (engine.getValue("[Channel" + deck + "]", "play") !== 1 || DJCi300.scratchButtonState) {
+            DJCi300._scratchEnable(deck);
+            DJCi300.scratchAction[deck] = DJCi300.kScratchActionScratch;
+        } else {
+            DJCi300.scratchAction[deck] = DJCi300.kScratchActionBend;
+        }
+    } else {
+        // Released the wheel.
+        engine.scratchDisable(deck);
+        DJCi300.scratchAction[deck] = DJCi300.kScratchActionNone;
+    }
+};
+
+// The touch action on the jog wheel's top surface while holding shift
+DJCi300.wheelTouchShift = function(channel, control, value, _status, _group) {
+    var deck = channel - 3;
+    // We always enable scratching regardless of button state.
+    if (value > 0) {
+        DJCi300._scratchEnable(deck);
+        DJCi300.scratchAction[deck] = DJCi300.kScratchActionSeek;
+    } else {
+        // Released the wheel.
+        engine.scratchDisable(deck);
+        DJCi300.scratchAction[deck] = DJCi300.kScratchActionNone;
+    }
+};
+
+// Scratching on the jog wheel (rotating it while pressing the top surface)
+DJCi300._scratchWheelImpl = function(deck, value) {
+    var interval = DJCi300._convertWheelRotation(value);
+    var scratchAction = DJCi300.scratchAction[deck];
+    if (scratchAction === DJCi300.kScratchActionScratch) {
+        engine.scratchTick(deck, interval * DJCi300.scratchScale);
+    } else if (scratchAction === DJCi300.kScratchActionSeek) {
+        engine.scratchTick(deck,
+            interval *  DJCi300.scratchScale *
+            DJCi300.scratchShiftMultiplier);
+    } else {
+        DJCi300._bendWheelImpl(deck, value);
+    }
+};
+
+// Scratching on the jog wheel (rotating it while pressing the top surface)
+DJCi300.scratchWheel = function(channel, control, value, _status, _group) {
+    var deck = channel;
+    DJCi300._scratchWheelImpl(deck, value);
+};
+
+// Seeking on the jog wheel (rotating it while pressing the top surface and holding Shift)
+DJCi300.scratchWheelShift = function(channel, control, value, _status, _group) {
+    var deck = channel - 3;
+    DJCi300._scratchWheelImpl(deck, value);
+};
+
+DJCi300._bendWheelImpl = function(deck, value) {
+    var interval = DJCi300._convertWheelRotation(value);
+    engine.setValue("[Channel" + deck + "]", "jog",
+        interval * DJCi300.bendScale);
+};
+
+// Bending on the jog wheel (rotating using the edge)
+DJCi300.bendWheel = function(channel, control, value, _status, _group) {
```

Comment:
Regarding all the merging of the jogwheel functions; are those necessary or mostly code cosmetic ? 

I ask because the same code is actually used in the DJControl Starlight #2061 (which has been accepted as official mapping), and honestly since Kerrick came up with all these improvements (my first draft was using the jog wheel code example from the wiki ), I wouldn't know where to start to make mergers without braking something.